### ISSUE TITLE
[Settings] Custom views - Handle outputed script tags

### ISSUE THREAD

**Issue by psealock** [woothemes/woocommerce#52975]
Sometimes extensions will place a `<script />` tag in the custom output of a settings field, such as in https://github.com/woocommerce/woocommerce/issues/52317. This script tag may include javascript or an external script. Make sure this is passed on in SettingsData to the client. Note that these scripts should be their own Setting element with type `script` (or something).

The proof of concept used `WP_HTML_Tag_Processor` for this.

https://github.com/woocommerce/woocommerce/blob/09cdb35a1ebbc422f87c5dea1795b0b7824d254e/plugins/woocommerce/includes/admin/settings/class-wc-settings-page.php#L182-L193

**Comment by chihsuan** [comment]
@psealock The [`WP_HTML_Tag_Processor`](https://developer.wordpress.org/reference/classes/wp_html_tag_processor/) doesn't provide a way to remove a tag. Using regex to remove the script tag might be possible. However, I think a better way would be to do this on the JS side. What do you think?  I think we could lazy-process the HTML when rendering the custom view.

The example code provided by AI is as follows:

```js
function processHTMLString( HTML ) {
  // Create parser and parse HTML
  const parser = new DOMParser();
  const doc = parser.parseFromString(html, 'text/html');
  
  // Find and process script tags
  const sectionSettingsData = [];
  const scriptTags = Array.from(doc.getElementsByTagName('script'));
  
  scripts.forEach(script => {
    if (script.getAttribute('type') === 'text/javascript') {
      sectionSettingsData.push({
        type: 'script',
        content: script.textContent
      });
      script.remove();
    }
  });

  return {
    processedHTML: doc.documentElement.outerHTML,
    extractedScripts: sectionSettingsData
  };
}
```

**Comment by psealock** [comment]
Yes, thats a good idea @chihsuan . The scripts will be appended/removed to the DOM by JS on render anyways. It would be good to gather the relevant tab/section as part of the data. This way, any scripts can be applied in the correct areas only.
